File: rgctl-tests/ecommerce-python/app/schemas/category.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


class CreateCategoryRequest(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    slug: str = Field(min_length=1, max_length=255)


class CategoryResponse(ORMModel):
    id: str
    name: str
    slug: str
    created_at: datetime

File: rgctl-tests/ecommerce-rust/src/dto/category.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateCategoryRequest {
    pub name: String,
    pub slug: String,
}

#[derive(Debug, Serialize)]
pub struct CategoryResponse {
    pub id: String,
    pub name: String,
    pub slug: String,
}
